Src/FluentAssertions/Equivalency/EnumEqualityStep.cs

```diff
@@ -56,5 +52,65 @@ public bool CanHandle(IEquivalencyValidationContext context, IEquivalencyAsserti
 
             return true;
         }
+
+        private static void HandleByValue(IEquivalencyValidationContext context)
+        {
+            decimal? subjectsUnderlyingValue = ExtractDecimal(context.Subject);
+            decimal? expectationsUnderlyingValue = ExtractDecimal(context.Expectation);
+
+            Execute.Assertion
+                .ForCondition(subjectsUnderlyingValue == expectationsUnderlyingValue)
+                .FailWith(() =>
+                {
+                    string subjectsName = EnumDescription(context.Subject, subjectsUnderlyingValue);
+                    string expectationName = EnumDescription(context.Expectation, expectationsUnderlyingValue);
+
+                    return new FailReason($"Expected enum to equal {expectationName} by value{{reason}}, but found {subjectsName}.");
```

Comment:
🤡 Consider using `{context:enum}` so that the caller C# code can be used to get the real name of the variable that is being asserted.

Reply:
Could you point me to some places in the code where I should look to understand that feature better? I guess `{context:enum}` would replace `"enum"`, wouldn't it? Any other considerations after that change, maybe some more tests?

Reply:
Nevermind, I took a look at existing tests and `MessageBuilder.SubstituteIdentifier`. Fixed.